inference-engine/src/transformations/src/transformations/op_conversions/convert_sequences_to_tensor_iterator.cpp

```diff
@@ -0,0 +1,583 @@
+            // create initial values for body_parameters in outer graph
+            // aggregated Y_h - concatenation of the last non-zero values for each batch
+            auto aggregated_Y_h = std::make_shared<ngraph::opset5::Constant>(H_body_param->get_element_type(),
+                                                                             H_body_param->get_shape(),
+                                                                             std::vector<float>(shape_size(H_body_param->get_shape()),
+                                                                                                0.f));
+            auto init_val_curr_iter = std::make_shared<ngraph::opset5::Constant>(seq_lengths.get_element_type(),
+                                                                                 ngraph::Shape{1},
+                                                                                 std::vector<int64_t>{1});
+            ngraph::copy_runtime_info(sequence, {aggregated_Y_h, init_val_curr_iter});
+
+            // set initial value and back edge for current iteration
+            tensor_iterator->set_merged_input(body_params.at(0), init_val_curr_iter, body_results.at(0));
+            // set initial value and back edge for aggregated H
+            tensor_iterator->set_merged_input(body_params.at(1), aggregated_Y_h, body_results.at(1));
+            H_out = tensor_iterator->get_function()->get_results()[1];
+        }
+        tensor_iterator->get_iter_value(H_out);
+        tensor_iterator->set_friendly_name(sequence->get_friendly_name());
+        if (enable_mask && is_reverse) {
+            auto reverse_seq_after = std::make_shared<opset5::ReverseSequence>(tensor_iterator->output(0), seq_lengths, 0, 2);
+            // Resolve a collision of names data nodes in CNN Network in Reverse case with mask.
+            /*
+             *   Before transformation (no collisions)
+             *   RNN/LSTM/GRU Sequence [rnn_name] -- (data_node: rnn_name.0) - > Result1
+             *                                    -- (data_node: rnn_name.1) - > Result2
+             *
+             *
+             *   After transformation (without identity, there are collisions):
+             *   We need to set rnn_name.0 to RevSequence to store result name.
+             *   TI [rnn_name] -- (DATA_NODE: rnn_name.0) --> RevSequence [rnn_name.0] -- (DATA_NODE: rnn_name.0) -> Result1
+             *                 -- (data_node: rnn_name.1) --> Result2
+             *
+             *
+             *   After transformation (with identity, no collisions):
+             *   TI has other_name, but it doesn't affect result names due TI is not connected to Results directly.
+             *   TI [other_name] -- (data_node: other_name.0) --> RevSequence [rnn_name.0] -- (data_node: rnn_name.0) -> Result1
+             *                   -- (data_node: other_name.1) --> Identity(rnn_name.1) -- (data_node: rnn_name.1) -> Result2
+             */
```

Comment:
Very complex logic! Thank you for the comments. Could you highlight this in the epic for outputs naming 37965?

Reply:
I have updated the description (Known issues list) in 37965, but I didn't create separate tickets.